test/unit/discovery_rule_test.rb

```diff
@@ -1,20 +1,59 @@
 require 'test_helper'
 
 class DiscoveryRuleTest < ActiveSupport::TestCase
+  setup do
+    @hg_org = FactoryGirl.create(:organization, :name => 'hgorg')
+    @hg_org.save!
+    @hostgroup = FactoryGirl.create(:hostgroup)
+    @hostgroup.organizations << @hg_org
+    @hostgroup.save!
+  end
 
   test "should be able to create a rule without entering hosts limit" do
-    rule = DiscoveryRule.new :name => "myrule", :search => "cpu_count > 1", :priority => "1", :hostgroup_id => "1"
+    rule = DiscoveryRule.new :name => "myrule",
+      :search => "cpu_count > 1",
+      :priority => "1",
+      :hostgroup_id => @hostgroup.id,
+      :organization_ids => [@hg_org.id]
     assert_valid rule
   end
 
   test "should be able to create a rule without entering priority" do
-    rule = DiscoveryRule.new :name => "myrule", :search => "cpu_count > 1", :max_count => "1", :hostgroup_id => "1"
+    rule = DiscoveryRule.new :name => "myrule",
+      :search => "cpu_count > 1",
+      :max_count => "1",
+      :hostgroup_id => @hostgroup.id,
+      :organization_ids => [@hg_org.id]
     assert_valid rule
   end
 
   test "should not be able to create a rule without a hostgroup" do
-    rule = DiscoveryRule.new :name => "myrule", :search => "cpu_count > 1"
+    rule = DiscoveryRule.new :name => "myrule",
+      :search => "cpu_count > 1",
+      :organization_ids => [@hg_org.id]
+    refute_valid rule
+  end
+
+  test "should enforce hostgroup organizations" do
+    skip unless SETTINGS[:organizations_enabled]
+    rule = DiscoveryRule.new :name => "myrule",
+      :search => "cpu_count > 1",
+      :hostgroup_id => @hostgroup.id
     refute_valid rule
+    assert_equal 'Host group organization hgorg must also be associated to the discovery rule', rule.errors[:organizations].first
```

Comment:
I'd use `@hg_org.name` instead of writing `hgorg` in case anyone ever changes the org and also it took me a second to figure out this was the name for the org above.